_posts/2022-08-23-luz-0-3/luz-0-3.Rmd

```diff
@@ -0,0 +1,206 @@
+date back to that version.
+
+## What's luz?
+
+Since luz is [relatively new
+package](https://blogs.rstudio.com/ai/posts/2021-06-17-luz/) we are
+starting this blog post with a quick recap of how luz works, if you
+already know what luz is, feel free to move to the next section.
+
+luz is a high-level API for torch that aims to encapsulate the training
+loop into a set of reusable pieces of code. It reduces the boilerplate
+code required to train a model with torch, avoids the error prone
+`zero_grad()` - `backward()` - `step()` sequence of calls, and also
+simplifies the process of moving data and models between CPUs and GPUs.
+
+With luz you can take your torch `nn_module()`, for example the
+two-layer perceptron defined below:
+
+```{r}
+modnn <- nn_module(
+  initialize = function(input_size) {
+    self$hidden <- nn_linear(input_size, 50)
+    self$activation <- nn_relu()
+    self$dropout <- nn_dropout(0.4)
+    self$output <- nn_linear(50, 1)
+  },
+  forward = function(x) {
+    x %>% 
+      self$hidden() %>% 
+      self$activation() %>% 
+      self$dropout() %>% 
+      self$output()
+  }
+)
+```
+
+and fit it to a specified a dataset with:
+
+```{r}
+fitted <- modnn %>% 
+  setup(
+    loss = nn_mse_loss(),
+    optimizer = optim_rmsprop,
+    metrics = list(luz_metric_mae())
+  ) %>% 
+  set_hparams(input_size = 50) %>% 
+  fit(
+    data = list(x_train, y_train),
+    valid_data = list(x_valid, y_valid),
+    epochs = 20
+  )
+```
+
+luz will automatically train your model on the GPU if it's available,
+display a nice progress bar during training, handle metric logging,
+while making sure evaluation on validation data is correctly performed
+(eg. disabling dropout).
+
+luz can be extended in many different layers of abstraction so you can
+gradually learn how it works as you need more advanced features in your
+project. For example, you can implement [custom
+metrics](https://mlverse.github.io/luz/reference/luz_metric.html),
+[callbacks](https://mlverse.github.io/luz/reference/luz_callback.html)
+or even customizing the [internal training
+loop](https://mlverse.github.io/luz/articles/custom-loop.html).
+
+If you want to get started with luz read the [getting
+started](https://mlverse.github.io/luz/articles/get-started.html)
+section in the website and navigate trough the [examples
+gallery](https://mlverse.github.io/luz/articles/examples/index.html).
+
+## What's new luz?
+
+### Learning rate finder
+
+In deep learning, finding a good learning rate is essential to be able
+to fit your model. If it's too low, you will need too many iterations
+for your loss to converge, and that might be impractical if your model
+takes too long to run. If it's too high, the loss can explode and you
+might never be able to minimize the loss.
+
+The `lr_finder()` function implements the ['Cyclical Learning Rates for
+Training Neural Networks'](https://arxiv.org/abs/1506.01186)
+([@smith2015]) popularized in the FastAI framework ([@howard2020]). It
```

Comment:
```suggestion
[@smith2015] popularized in the FastAI framework [@howard2020]. It
```

Reply:
I think this should evaluate correctly, without the double parens. Can you please check when rendering?